src/components/media-player/dialog-join-media-players.ts

```diff
@@ -0,0 +1,201 @@
+import { customElement, property, state } from "lit/decorators";
+import type { HassEntity } from "home-assistant-js-websocket";
+import { fireEvent } from "../../common/dom/fire_event";
+import { haStyleDialog } from "../../resources/styles";
+import type { HomeAssistant } from "../../types";
+import "../ha-alert";
+import "../ha-dialog";
+import "../ha-button";
+import "../ha-dialog-header";
+import "./ha-media-player-toggle";
+import "../../panels/lovelace/components/hui-marquee";
+import type { JoinMediaPlayersDialogParams } from "./show-join-media-players-dialog";
+import { computeStateName } from "../../common/entity/compute_state_name";
+import { supportsFeature } from "../../common/entity/supports-feature";
+import {
+  type MediaPlayerEntity,
+  MediaPlayerEntityFeature,
+  mediaPlayerJoin,
+  mediaPlayerUnjoin,
+} from "../../data/media-player";
+import { extractApiErrorMessage } from "../../data/hassio/common";
+import type { EntityRegistryDisplayEntry } from "../../data/entity_registry";
+
+@customElement("dialog-join-media-players")
+class DialogJoinMediaPlayers extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _entityId?: string;
+
+  @state() private _groupMembers!: string[];
+
+  @state() private _selectedEntities!: string[];
+
+  @state() private _submitting?: boolean;
+
+  @state() private _error?: string;
+
+  public showDialog(params: JoinMediaPlayersDialogParams): void {
+    this._entityId = params.entityId;
+
+    const stateObj = this.hass.states[params.entityId] as
+      | MediaPlayerEntity
+      | undefined;
+
+    this._groupMembers =
+      stateObj?.attributes.group_members?.filter(
+        (entityId) => entityId !== params.entityId
+      ) || [];
+
+    this._selectedEntities = this._groupMembers;
+  }
+
+  public closeDialog() {
+    this._entityId = undefined;
+    this._selectedEntities = [];
+    this._submitting = false;
+    this._error = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render() {
+    if (!this._entityId) {
+      return nothing;
+    }
+
+    const entityId = this._entityId;
+    const stateObj = this.hass.states[entityId] as HassEntity | undefined;
+    const name = (stateObj && computeStateName(stateObj)) || entityId;
+
+    return html`
+      <ha-dialog
+        open
+        scrimClickAction
+        escapeKeyAction
+        flexContent
+        .heading=${name}
+        @closed=${this.closeDialog}
+      >
+        <ha-dialog-header show-border slot="heading">
+          <ha-icon-button
+            .label=${this.hass.localize("ui.common.close")}
+            .path=${mdiClose}
+            dialogAction="close"
+            slot="navigationIcon"
+          ></ha-icon-button>
+          <span slot="title"
+            >${this.hass.localize("ui.card.media_player.media_players")}</span
+          >
+          <ha-button slot="actionItems" @click=${this._selectAll}>
```

Comment:
Based on the UI mocks, we'd like the button to be vertically centered with the rest of the items. From what I can tell, the action items container is not vertically centered. This seems to be quite deep in the component hierarchy, so I'm unsure what would be the best solution.

I'm basically after this:

```css
.header-bar {
  align-items: center;
}
```

The current style is to align it to the flex start. Why, I don't know.

Reply:
It only seems to be used to display icons. I've left a css suggestion that you can apply to place it vertically.